### PR TITLE
feat: Trace jobs over a run

### PR DESCRIPTION
**Pull request by lforst**
This PR will set trace IDs on the generated transactions based on Github `run_id` and the `run_attempt`.

This should create connected traces for all job transaction within one workflow run.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -40,11 +40,11 @@ def uuid_list():
     return [
         "5ae279acd9824cbfa85042013cfbf8b7",
         "a401d83c7ec0495f82a8da8d9a389f5b",
-        "81b0bea6aedf4817a74d32d706908df2",
+        "4d4d3477c624836b1b3a3729a7de688a",
         "0726fb4a2341477cbfa36ebd830bd8e3",
         "a7776ed12daa449bb0642e9ea1bb4152",
         "6e147ff372f9498abb5749a1210d8e0a",
-        "498cceede5f14fd991778f007b237803",
+        "a401d83c7ec0495f82a8da8d9a389f5b",
```
Comment (lforst) [inline tests/conftest.py:47]: Honestly no idea why these needed to be changed

File: tests/test_github_sdk.py [hunk 2]
```diff
@@ -120,7 +120,7 @@ def test_send_trace(
         "Content-Type": "application/x-sentry-envelope",
         "Content-Encoding": "gzip",
         "X-Sentry-Auth": f"Sentry sentry_key=foo,sentry_client=gha-sentry/0.0.1,sentry_timestamp={now},sentry_version=7",
-        "Content-Length": "701",
+        "Content-Length": "693",
```
Comment (lforst) [inline tests/test_github_sdk.py:123]: better gzip?

File: tests/conftest.py (hunk 1 above)
Comment (armenzg) [reply tests/conftest.py:47]: Don't worry about it. I did not write the tests correctly.

File: tests/test_github_sdk.py (hunk 2 above)
Comment (armenzg) [reply tests/test_github_sdk.py:123]: Possibly the trace ID is a little shorter.

**Review by armenzg — APPROVED** [review]
My apologies you had to deal with the ids and poorly written tests.
Thanks for the fix!

**Pull request merged**